sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -2965,4 +2965,21 @@ class SQLQuerySuite extends QueryTest with SharedSQLContext {
       }
     }
   }
+
+  test("Aggregate should not propagate constraints containing aggregate expressions") {
```

Comment:
Nice find. I agree, it'd be good to put this in `ConstraintPropagationSuite` and explicitly check for the inferred constraints.